### ISSUE TITLE
Request for a new API: Update captions by transcription ID

### ISSUE THREAD

**Issue by yiyanglin98** [classtranscribe/WebAPI#134]
The current caption updating API in `/api/Captions` can only update a single caption line with one API call, which is insufficient for frontend issue [#103](https://github.com/classtranscribe/FrontEnd/issues/103) for two reasons.

1. Bulk editing of captions needs to update multiple caption lines (text, start time and end time) by transcription ID, with a single API call.
2. User should be able to upload new caption document by transcription ID.

Therefore, it would be great to have a new API that can do PUT requests by transcription ID, to update/create captions efficiently.

**Comment by angrave** [comment]
Agreed! How should it work? Replace only captions that are different?Or does the frontend only provide inserts? How are soft deletes handled? Some thought needs to be put into creating a spec for this.

**Comment by yiyanglin98** [comment]
> Agreed! How should it work? Replace only captions that are different?Or does the frontend only provide inserts? How are soft deletes handled? Some thought needs to be put into creating a spec for this. So why not draft what it should do...

Sure. I will add more details in the next few days.

Also, I am not sure why soft deletion is necessary here. Is that for data recovery? Do we need to offer an interface for the users to undo their deletion? Or there are other purposes for soft deletion here, e.g. counting the deleted captions.

**Comment by angrave** [comment]
Undo. We actually want a full revision history not just soft delete: e.g. want to be able to review changes made by a particular (or all) contributors or changes since date. Be able roll-back or undo a set of changes.

Also changes in main transcription suggest where changes should be made in translations

**Comment by tdy** [comment]
Will WebVTT just piggyback off this API spec, or are there other needs/considerations for a WebVTT API?

**Comment by angrave** [comment]
Yes, WebVTT uploaded to the frontend could piggyback off this API. So the ability to upload a vtt into the frontend would be useful.

(However we will need to support extracting captions from mp4 / upstream sources - so I imagine their backend will need some native WebVTT support to.)
